### PR TITLE
Restored the one-hot functionality from an older branch.

### PR DESCRIPTION
**Pull request by vpatryshev**
Michal, you already saw this pr; restored.

### CODE REVIEW

**Comment by vpatryshev** [comment]
@h2o-ops please test!

**Comment by vpatryshev** [comment]
@h2o-ops please test!

File: h2o-scala/src/main/scala/water/udf/MoreColumns.scala [hunk 1]
```diff
@@ -130,16 +146,17 @@ object MoreColumns extends DataColumns {
...
+  def unfoldingColumn[X, Y]
+  (f: X => Iterable[Y], source: Column[X], width: Int): UnfoldingColumn[X, Y] = {
     new UnfoldingColumn[X, Y](fu1[X, Y](f), source, width)
   }
 
-//  def unfoldingFrame[X, Y](f: X => Iterable[Y], source: Column[X], width: Int) = {
-//    new UnfoldingFrame[X, Y](fu1[X, Y](f), source, width)
-//  }
-   
+  //  def unfoldingFrame[X, Y](f: X => Iterable[Y], source: Column[X], width: Int) = {
```
Comment (mmalohlava) [inline h2o-scala/src/main/scala/water/udf/MoreColumns.scala]: Probably not necessary to keep it here.

File: h2o-core/src/main/java/water/fvec/Frame.java [hunk 2]
```diff
@@ -1588,4 +1588,11 @@ public int getCurrentRowSize() throws IOException {
   /** Sort rows of a frame, using the set of columns as keys.
    *  @return Copy of frame, sorted */
   public Frame sort( int[] cols ) { return Merge.sort(this,cols); }
+
+  public void checkVecs(String message) {
+    for (Vec vec : vecs()) if (DKV.get(vec._key)== null) {
+      throw new IllegalStateException(message + ": very bad vec " + vec + ", total disaster.");
```
Comment (vpatryshev) [inline h2o-core/src/main/java/water/fvec/Frame.java]: Done.

File: h2o-core/src/main/java/water/udf/MatrixFrame.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
+package water.udf;
+
+import water.fvec.Frame;
+import water.fvec.Vec;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Single column frame that knows its data type and can materialize
```
Comment (vpatryshev) [inline h2o-core/src/main/java/water/udf/MatrixFrame.java]: Right; fixed.

File: h2o-core/src/main/java/water/udf/SingleColumnFrame.java [hunk 4]
```diff
@@ -5,60 +5,66 @@
...
+  SingleColumnFrame(BaseFactory<DataType, ColumnType> factory, long length, Function<Long, DataType> function) {
     super();
     this.factory = factory;
     this.length = length;
     this.function = function;
   }
 
-  public static <X> TypedFrame<X> forColumn(final BaseFactory<X> factory, final Column<X> column) {
-    return new TypedFrame<X>(factory, column.size(), column) {
+  public static
```
Comment (vpatryshev) [inline h2o-core/src/main/java/water/udf/SingleColumnFrame.java]: Fixed

File: h2o-core/src/main/java/water/util/FrameUtils.java [hunk 5]
```diff
@@ -77,6 +83,17 @@ public static Frame categoricalEncoder(Frame dataset, String[] skipCols, Model.P
       case Enum:
       case OneHotInternal:
         return dataset; //leave as is - most algos do their own internal default handling of enums
+// Commented out until further notice.
```
Comment (vpatryshev) [inline h2o-core/src/main/java/water/util/FrameUtils.java]: I don't mind kicking it out.

File: h2o-scala/src/main/scala/water/udf/MoreColumns.scala (hunk 1 above)
Comment (vpatryshev) [inline h2o-scala/src/main/scala/water/udf/MoreColumns.scala]: Sure; removed.

File: h2o-core/src/main/java/water/udf/SingleColumnFrame.java (hunk 4 above)
Comment (vpatryshev) [inline h2o-core/src/main/java/water/udf/SingleColumnFrame.java]: Just did.

**Comment by vpatryshev** [comment]
@h2o-ops please test!

**Comment by mmalohlava** [comment]
@vpatryshev looks ok, just two things:
  - there are conflicts
  - could you prepare package level documentation with the main concepts.

**Pull request closed without merging**